src/natcap/invest/recreation/recmodel_server.py

```diff
@@ -687,49 +687,54 @@ def _calc_poly_pud(
     LOGGER.info('local qt load took %.2fs', time.time() - start_time)
 
     aoi_vector = gdal.OpenEx(aoi_path, gdal.OF_VECTOR)
-    aoi_layer = aoi_vector.GetLayer()
-
-    for poly_id in iter(poly_test_queue.get, 'STOP'):
-        poly_feat = aoi_layer.GetFeature(poly_id)
-        poly_geom = poly_feat.GetGeometryRef()
-        poly_wkt = poly_geom.ExportToWkt()
-        try:
-            shapely_polygon = shapely.wkt.loads(poly_wkt)
-        except Exception:  # pylint: disable=broad-except
-            # We often get weird corrupt data, this lets us tolerate it
-            LOGGER.warn('error parsing poly, skipping')
-            continue
-
-        poly_points = local_qt.get_intersecting_points_in_polygon(
-            shapely_polygon)
-        pud_set = set()
-        pud_monthly_set = collections.defaultdict(set)
-
-        for point_datetime, user_hash, _, _ in poly_points:
-            if date_range[0] <= point_datetime <= date_range[1]:
-                timetuple = point_datetime.tolist().timetuple()
-
-                year = str(timetuple.tm_year)
-                month = str(timetuple.tm_mon)
-                day = str(timetuple.tm_mday)
-                pud_hash = str(user_hash) + '%s-%s-%s' % (year, month, day)
-                pud_set.add(pud_hash)
-                pud_monthly_set[month].add(pud_hash)
-                pud_monthly_set["%s-%s" % (year, month)].add(pud_hash)
-
-        # calculate the number of years and months between the max/min dates
-        # index 0 is annual and 1-12 are the months
-        pud_averages = [0.0] * 13
-        n_years = (
-            date_range[1].tolist().timetuple().tm_year -
-            date_range[0].tolist().timetuple().tm_year + 1)
-        pud_averages[0] = len(pud_set) / float(n_years)
-        for month_id in range(1, 13):
-            monthly_pud_set = pud_monthly_set[str(month_id)]
-            pud_averages[month_id] = (
-                len(monthly_pud_set) / float(n_years))
-
-        pud_poly_feature_queue.put((poly_id, pud_averages, pud_monthly_set))
+    if aoi_vector:
```

Comment:
This whole block is just to guard against that `aoi_vector` being None right and then getting an indention?

Reply:
@richpsharp , yes there's the `if` and also I added the try/except for the `AttributeError`. Basically covering all the cases that came up recently that are collected in #304